lib/conduit_async_ssl.mli

```diff
@@ -33,7 +33,7 @@ val ssl_connect :
   ?verify:(Ssl.Connection.t -> bool Deferred.t) ->
   Reader.t ->
   Writer.t ->
-  (Reader.t * Writer.t) Deferred.t
+  (Reader.t * Writer.t) Deferred.Or_error.t
```

Comment:
> The |verify| function could raise, for example. And indeed the
> |Reader.of_pipe| can raise if there is no fd left.

About verify, this is something that we should definitely reflect in the
docs. Either that it shouldn't raise or that we'll catch whatever it
raises and return it as Or_error.

> 
> I'm not certain anymore what's the best thing to do. I think best would
> be wrap everything in a big try_with or do not return Or_error.t types.
> What do you think?

My biggest concern is now actually backwards compatibility. But it seems
like you might be the only user :)

https://github.com/search?q=conduit_async_ssl&type=Code&utf8=%E2%9C%93

Out of curiousity, is it possible to just use Conduit_async for your use
case?

Anyway, let's see what the other maintainers say about this interface
change.

@avsm @samoht @dsheets

I will cherry pick the error swallowing fix regardless of course.